RoboJackets/igvc-software
# Merged waypoint_source and action_server into navigation_client

# Description

This PR does the following:
- Merges `waypoint_source` and `action_server` functionality into the new `navigation_client`
- Removes `waypoint_source` and `action_server`
- Changes the `navigation_simulation` and `simulation` launch files to use `navigation_client`

Goes toward fixing #596 and #520 

# Testing steps (If relevant)
## Test reading from file
1. Make sure `navigation_client` launch file has the `reading_from_file` param set to `true`
2. Run `roslaunch igvc_gazebo qualification.launch` and `roslaunch igvc_navigation navigation_simulation.launch` 
Expectation: Robot goes to the waypoint.

## Test getting goals from rviz
1. Change the `reading_from_file` parameter to `false`
2. Open up the simulation and rviz, and set a goal pose with rviz.
Expectation: Robot goes to the goal set by rviz.


# Self Checklist
